### PR TITLE
 fix FA2 when using quantization for remaining models

### PR DESCRIPTION
**Pull request by susnato**
# What does this PR do?

Replicates this PR #28203 for remaining models.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/models/distilbert/modeling_distilbert.py [hunk 1]
```diff
@@ -322,11 +322,13 @@ def reshape(x: torch.Tensor) -> torch.Tensor:
...
         if query_states.dtype == torch.float32:
+            if torch.is_autocast_enabled():
+                target_dtype = torch.get_autocast_gpu_dtype()
             # Handle the case where the model is quantized
-            if hasattr(self.config, "_pre_quantization_dtype"):
+            elif hasattr(self.config, "_pre_quantization_dtype"):
                 target_dtype = self.config._pre_quantization_dtype
             else:
-                target_dtype = self.q_lin.weight.dtype
+                target_dtype = self.q_proj.weight.dtype
```
Comment (ArthurZucker) [inline src/transformers/models/distilbert/modeling_distilbert.py:331]: q_proj does not exist
```suggestion
                target_dtype = self. q_lin.weight.dtype
```

**Review by ArthurZucker — APPROVED** [review]
Thanks 🤗

**Comment by susnato** [comment]
Done! @ArthurZucker

File: src/transformers/models/distilbert/modeling_distilbert.py [hunk 2]
```diff
@@ -322,11 +322,13 @@ def reshape(x: torch.Tensor) -> torch.Tensor:
...
         if query_states.dtype == torch.float32:
+            if torch.is_autocast_enabled():
+                target_dtype = torch.get_autocast_gpu_dtype()
             # Handle the case where the model is quantized
-            if hasattr(self.config, "_pre_quantization_dtype"):
+            elif hasattr(self.config, "_pre_quantization_dtype"):
                 target_dtype = self.config._pre_quantization_dtype
             else:
-                target_dtype = self.q_lin.weight.dtype
+                target_dtype = self. q_lin.weight.dtype
```
Comment (ArthurZucker) [inline src/transformers/models/distilbert/modeling_distilbert.py:331]:
```suggestion
                target_dtype = self.q_lin.weight.dtype
```
sorry my suggestion had a typo it seems!

**Comment by susnato** [comment]
CI is green now! @ArthurZucker

**Comment by ArthurZucker** [comment]
Thanks for the contribution! FYI @pacman100 and @younesbelkada

**Pull request merged**